Scripts/Player/Player.gd

```diff
@@ -15,32 +15,37 @@ var path: Array = []                        ## Array to hold the calculated path
 var path_index: int = 0                     ## Current index in the path array.
 var highlight_path: PackedVector2Array = [] ## Array to hold the positions to highlight on the path.
 var selected_object: InteractableObject     ## Reference to the interactable object currently selected by the player.
+var camera: Camera2D                        ## Refrence to Camera
 var action: bool = false:                   ## Flag indicating if the player is currently performing an action.
 	set(value):
 		action = value
 		queue_redraw()
 
+# --- Private Properties ---
+@onready var _actions: int = actions
+
 # --- Signals ---
 signal moved(position: Vector2i)            ## Signal emitted when the player’s position is updated.
-
+signal turn_ended                           ## Signal emitted when the player's turn ends
 # --- Built-in Callbacks ---
 # Called when the node is added to the scene. Sets up camera limits and connects the resize signal.
 func _ready() -> void:
-	
+	camera = get_child(UtilityFunctions.get_from_type(get_children(), "Camera2D"))
 	global.player = self
-	$Camera2D.limit_left = global.map.get_used_rect().position.x * 128
-	$Camera2D.limit_top = global.map.get_used_rect().position.y * 128
-	$Camera2D.limit_right = (global.map.get_used_rect().position.x + global.map.get_used_rect().size.x) * 128
-	$Camera2D.limit_bottom = (global.map.get_used_rect().position.y + global.map.get_used_rect().size.y) * 128
+	camera.limit_left = global.map.get_used_rect().position.x * 128
```

Comment:
This is a little confusing, could you document it?

Reply:
sure! but thats nothing new. (that was already there from a previous PR)